sui/utils.js

```diff
@@ -77,15 +77,20 @@ const deployPackage = async (packageName, client, keypair, options = {}) => {
     return { packageId, publishTxn };
 };
 
-const findPublishedObject = (published, packageName, contractName) => {
-    const packageId = published.packageId;
-    return published.publishTxn.objectChanges.find((change) => change.objectType === `${packageId}::${packageName}::${contractName}`);
+const getObjectIdsByObjectTypes = (txn, objectTypes) => {
+    const objectIds = [];
+
+    for (const objectType of objectTypes) {
+        objectIds.push(txn.objectChanges.find((change) => change.objectType === objectType).objectId);
+    }
+
+    return objectIds;
 };
```

Comment:
```suggestion
const getObjectIdsByObjectTypes = (txn, objectTypes) => 
  objectTypes.map((objectType) => {
    const objectId = txn.objectChanges.find(change => change.objectType === objectType)?.objectId;
    if (!objectId) {
      throw new Error(`No object found for type: ${objectType}`);
    }
  });
```
handle undefined cases. And use `map` to make it more functional